opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/aggregation/ExpressionAggregationScript.java

```diff
@@ -42,7 +49,22 @@ public ExpressionAggregationScript(
 
   @Override
   public Object execute() {
-    return expressionScript.execute(this::getDoc, this::evaluateExpression).value();
+    var expr = expressionScript.execute(this::getDoc, this::evaluateExpression);
+    if (expr.type() instanceof OpenSearchDataType) {
+      return expr.value();
+    }
+    switch ((ExprCoreType)expr.type()) {
+      case TIME:
+        // workaround for session context issue
+        // TODO remove once fixed
```

Comment:
Can you explain what the issue is? Link to any relevant github issues or PRs?

Reply:
Fixed in af13d834d.